### PR TITLE
Allow dropdown source for all string operators in dashboards

### PR DESCRIPTION
**Pull request by ranquild**
How to verify:
- Add string/contains parameter to the dashboard and connect it to a field
- Set "dropdown" options in the sidebar for this parameter
- Make sure you can select field values in the widget for this parameter
- Custom value source options should also work

Screenshot 2024-03-06 at 18 38 27
Screenshot 2024-03-06 at 18 38 33

### CODE REVIEW

**Review by cdeweyx — APPROVED** [review]
LGTM

File: frontend/src/metabase-lib/parameters/utils/parameter-source.unit.spec.ts [hunk 1]
```diff
@@ -30,6 +31,54 @@ const searchField = metadata.field(SEARCH_FIELD_ID) as Field;
...
+      (operator, queryType) => {
+        const parameter = createMockUiParameter({
+          type: `${type}/${operator}`,
+        });
+        expect(getQueryType(parameter)).toBe(queryType);
+      },
+    );
+  });
+
+  it('should return "none" query type by default for legacy parameter types', () => {
```
Comment (kamilmielnik) [inline frontend/src/metabase-lib/parameters/utils/parameter-source.unit.spec.ts:74]: The name of the test does not match the assertion (`none` vs `list`).

**Review by kamilmielnik — COMMENTED** [review]
Works well :+1: 
Left a comment inline.

File: frontend/src/metabase-lib/parameters/utils/parameter-source.unit.spec.ts (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase-lib/parameters/utils/parameter-source.unit.spec.ts]: Fixed

**Review by kamilmielnik — APPROVED** [review]

**Pull request merged**